transcom/mymove
# MB-5342 TXO see usmc moves

## Description

> For this one we’re just worried about making sure the branch dropdown on the front end is able to adapt to the backend work that was done for this in MB-4692: TOO & TIO can be assigned to handle USMC moves and see USMC moves in their queue (backend)
> 
> For this we will likely need to look at what we’re returning from the API side for the office user info and expand it to include what we need to handle these cases appropriately.
> 
> For non-USMC office users we should remove the USMC filter option from the branch drop down
> 
> As a USMC TOO/TIO, I want to be able to handle all USMC moves, so that…
> 
> Given that I am a TOO/TIO assigned to the USMC GBLOC
>  When I look at my move queue
>  Then I see all moves for USMC service members
> Given that I am *not* a TOO/TIO assigned to the USMC GBLOC
>  When I look at my move queue
>  Then I *don't* see any moves for USMC service members
>  
> 
> USTC is in the process of establishing the USMC PPSO Queue in MilMove.
> 
> GBLOC:  USMC
> Location:  Camp LeJeune, NC
> PPSO POC:  (need name, phone, email, etc.)
> Name for new record should be: Camp LeJeune (USMC)

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?

- Did a bit of work to refactor old payload code to make it more reusable 
- Added an new user that has both TOO/TIO roles but with the gbloc USMC to test
- Opted to load the user with react-query instead of trying to hook up with redux to avoid over complicating the components

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

**To test non-USMC gblocs**
1. Login as `too_tio_role@office.mil`, local
2. Click on branch dropdown to check that `Marine Corps` is not added
3. Switch roles to check the same

**To test USMC gblocs**
1. Login as `too_tio_role_usmc@office.mil`, local, new from devseed
2. Check to see branch filtering is removed
3. Switch roles to check the same

## Code Review Verification Steps


 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-5342) for this change

## Screenshots

User not of USMC gbloc, Marine Corps removed from options
![image](https://user-images.githubusercontent.com/1522549/99829856-b4a3bc80-2b11-11eb-99f7-974026aedebe.png)

User of USMC gbloc, branch filtering is removed
![image](https://user-images.githubusercontent.com/1522549/100019398-d19af280-2d92-11eb-86bb-200830f7edb0.png)
